libkodimote/kodi.cpp

```diff
@@ -498,14 +512,28 @@ void Kodi::setVolume(int volume)
 
     } else {
         if(volume != m_volume) {
-            QVariantMap map;
-            map.insert("volume", volume);
-            KodiConnection::sendCommand("Application.SetVolume", map);
-
+            requestVolume(volume);
         }
     }
 }
 
+void Kodi::requestVolume(int volume) {
+    m_volumePending = volume;
+    if (!m_volumeChangeOutstanding) {
+        m_volumeChangeOutstanding = true;
+        QVariantMap map;
+        map.insert("volume", volume);
+        KodiConnection::sendCommand("Application.SetVolume", map);
```

Comment:
In that case you could probably drop `m_volumePending`, and `m_volumeOutstanding`, and add an `int m_volumeId`, which is `-1` if there is no outstanding `SetVolume` request, and has the id of the last `SetVolume` (or `Application.GetProperties`, for volume) call when there is a volume request. `volumeReceived` should then check if the id of the response is that to the last volume request (`m_volumeId`) and update the volume if it is. Handling the `OnVolumeChanged` notification should then only be done when `m_volumeId` is `-1` (otherwise we still expect a reply with the volume).